File: Learn_Android/getWallpaer/app/src/main/java/com/example/sunkai/getwallpaer/Adapter/RecyclerViewAdapter.java
package com.example.sunkai.getwallpaer.Adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.sunkai.getwallpaer.Data.imageData;
import com.example.sunkai.getwallpaer.R;

import java.util.List;

/*
 * Created by sunkai on 2018/1/26.
 */

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {


    private Context context;
    private List<imageData> datas;
    public RecyclerViewAdapter(final Context context, final List<imageData> datas){
        this.context=context;
        this.datas=datas;
    }
    @Override
    public RecyclerViewAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(context).inflate(R.layout.item_layout,parent,false);
        return new ViewHolder(view);
    }
    @Override
    public void onBindViewHolder(RecyclerViewAdapter.ViewHolder holder, int position) {
        imageData data=datas.get(position);
        holder.copyRight.setText(data.getCopyright());
        Glide.with(context).load(data.getUrl()).into(holder.image);

    }

    @Override
    public int getItemCount() {
        return datas.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{
        TextView copyRight;
        ImageView image;
        public ViewHolder(View view){
            super(view);
            copyRight=view.findViewById(R.id.item_title);
            image=view.findViewById(R.id.item_image);
        }
    }
}

File: Learn_Android/MyApplication/app/src/main/java/com/example/sunkai/myapplication/Adapter/RecyclerViewAdapter.kt
package com.example.sunkai.myapplication.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.sunkai.myapplication.Data.imageData
import com.example.sunkai.myapplication.R

/*
 * Created by sunkai on 2018/1/26.
 */
class RecyclerViewAdapter(val context: Context, val datas:List<imageData>):RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val view=LayoutInflater.from(context).inflate(R.layout.item_layout,parent,false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return datas.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val data=datas[position]
        holder.copyRight.text=data.copyright
        data.ok
        Glide.with(context).load(data.url).into(holder.image)
    }

    class ViewHolder(view: View):RecyclerView.ViewHolder(view){
        val copyRight:TextView
        val image:ImageView
        init {
            copyRight=view.findViewById<TextView>(R.id.item_title)
            image=view.findViewById<ImageView>(R.id.item_image)
        }
    }
}